src/CsvHelper/Configuration/ConfigurationFunctions.cs

```diff
@@ -225,27 +241,30 @@ public static string GetDynamicPropertyName(GetDynamicPropertyNameArgs args)
 			if (lineDelimiterCounts.Count > 1)
 			{
 				// The last line isn't complete and can't be used to reliably detect a delimiter.
-				lineDelimiterCounts.Remove(lineDelimiterCounts.Last());
+				lineDelimiterCounts.RemoveAt(lineDelimiterCounts.Count - 1);
 			}
 
-			// Rank only the delimiters that appear on every line.
+			var lineCount = lineDelimiterCounts.Count;
+
+			// Select only the delimiters that appear on every line.
 			var delimiters =
 			(
 				from counts in lineDelimiterCounts
 				from count in counts
 				group count by count.Key into g
-				where g.All(x => x.Value > 0)
-				let sum = g.Sum(x => x.Value)
-				orderby sum descending
 				select new
 				{
 					Delimiter = g.Key,
-					Count = sum
-				}
-			).ToList();
+					Count = g.Sum(x => x.Value),
+					LineCount = g.Count()
+				} into t
+				where t.LineCount == lineCount
+				orderby t.Count descending
+				select t
+			);
```

Comment:
Without the `ToList` this LINQ could be enumerated twice, was that intentional?

Reply:
Fixed.